public/js/lib/leaflet/extends/L.Google.js

```diff
@@ -1,185 +1,12 @@
-                    keyboardShortcuts: false,
-                    draggable: false,
-                    disableDoubleClickZoom: true,
-                    scrollwheel: false,
-                    streetViewControl: false
-                });
-
-                var _this = this;
-                this._reposition = google.maps.event.addListenerOnce(map, "center_changed",
-                    function () {
-                        _this.onReposition();
-                    });
-
-                map.backgroundColor = '#ff0000';
-                this._google = map;
-            },
-
-            _resetCallback: function (e) {
-                this._reset(e.hard);
-            },
-
-            _reset: function (clearOldContainer) {
-                this._initContainer();
-            },
-
-            _update: function () {
-                if (!this._google) return;
-                this._resize();
-
-                var bounds = this._map.getBounds();
-                var ne = bounds.getNorthEast();
-                var sw = bounds.getSouthWest();
-                var google_bounds = new google.maps.LatLngBounds(
-                    new google.maps.LatLng(sw.lat, sw.lng),
-                    new google.maps.LatLng(ne.lat, ne.lng)
-                );
-                var center = this._map.getCenter();
-                var _center = new google.maps.LatLng(center.lat, center.lng);
-
-                this._google.setCenter(_center);
-                this._google.setZoom(this._map.getZoom());
-                //this._google.fitBounds(google_bounds);
-            },
-
-            _resize: function () {
-                var size = this._map.getSize();
-                if (this._container.style.width == size.x &&
-                    this._container.style.height == size.y)
-                    return;
-                this.setElementSize(this._container, size);
-                this.onReposition();
-            },
-
-            onReposition: function () {
-                if (!this._google) return;
-                google.maps.event.trigger(this._google, "resize");
-            }
-        });
-
-        L.Google.asyncWait = [];
-        L.Google.asyncInitialize = function () {
-            var i;
-            for (i = 0; i < L.Google.asyncWait.length; i++) {
-                var o = L.Google.asyncWait[i];
-                o._ready = true;
-                if (o._container) {
-                    o._initMapObject();
-                    o._update();
-                }
-            }
-            L.Google.asyncWait = [];
-        }
-    })(window.google, L);
+define(['Params', 'leaflet', 'leaflet-plugins/Google'], function (P, L) {
+    let keyParam = P.settings.publicApiKeys.googleMaps.length ? '&key=' + P.settings.publicApiKeys.googleMaps : '';
+    let url = (location.protocol || 'http:') + '//maps.googleapis.com/maps/api/js?v=weekly&region=RU' + keyParam;
+    // Load Google Maps API library asynchronously.
+    require([ 'async!' + url]);
```

Comment:
Note for reviewer: `L.Google` got functionality to wait for libs availability using promise, so (unlike in Yandex case) we don't need to do anything special here apart of request libs loading and return layer class.